src/Components/ActionsBuilder/SupportedActions/SetPermissions/SetPermissionsEditor.tsx

```diff
@@ -99,66 +90,47 @@ const Permissions: React.FC<ActionEditorProps> = ({
   }, [tokens, tokenAddress]);
   // eslint-disable-next-line react-hooks/exhaustive-deps
 
-  // function signature
   const updateFunctionSignature = (value: string) => {
     if (!value || value === '')
-      return setValue('functionSignature', [ANY_FUNC_SIGNATURE]);
+      return setValue('functionSignature', ANY_FUNC_SIGNATURE);
 
     // If the value already is encoded
     if (value.substring(0, 2) === '0x')
-      return setValue('functionSignature', [value]);
+      return setValue('functionSignature', value);
 
     // if the value is the name of the function
     const functionSignature = web3.eth.abi.encodeFunctionSignature(value);
-    return setValue('functionSignature', [functionSignature]);
+
+    return setValue('functionSignature', functionSignature);
   };
 
   const handleFunctionNameChange = (value: string) => {
     setValue('functionName', value);
     updateFunctionSignature(value);
   };
 
-  const bigNumberMaxUINT = BigNumber.from(MAX_UINT);
-  const handleToggleMaxValueChange = () => {
-    // toggle on
-    if (!maxValueToggled) setValue('amount', bigNumberMaxUINT);
-    // toggle off
-    if (maxValueToggled) setValue('amount', BigNumber.from(tempAmount));
-    setMaxValueToggled(!maxValueToggled);
-  };
-
-  const handleToggleAnyAddressChange = () => {
-    // toggle on
-    if (!anyAddressToggled) {
-      setValue('toAddress', ANY_ADDRESS);
-    }
-    // toggle off
-    else {
-      setValue('toAddress', tempToAddress);
-    }
-    setAnyAddressToggled(!anyAddressToggled);
-    rest.trigger('toAddress');
-  };
+  const submitAction = (values: FormValues) => {
+    const bigNumberMaxUINT = BigNumber.from(MAX_UINT);
+    const isAssetTransferCall = activeTab === TABS.ASSET_TRANSFER;
 
-  const submitAction = values => {
-    const newcall = {
+    const newCall: DecodedCall = {
       ...decodedCall,
-      ...(activeTab === TABS.FUNCTION_CALL && {
-        functionName: values.functionName,
-      }), // Spread only if is functionCall tab
       args: {
         ...decodedCall.args,
-        ...(activeTab === TABS.ASSET_TRANSFER && {
-          asset: [values.tokenAddress],
-        }), // Spread only if is assetTransfer tab
-        ...(activeTab === TABS.FUNCTION_CALL && {
-          functionSignature: [values.functionSignature],
-        }), // Spread only if is functionCall tab
-        valueAllowed: [values.amount],
-        to: [values.toAddress],
+        to: isAssetTransferCall ? values.tokenAddress : values.toAddress,
+        valueAllowed: isAssetTransferCall ? bigNumberMaxUINT : values.amount,

```

Comment:
We should hardcode this as 0 for now if `isAssetTransferCall ` is true since it is setting native asset transfer value allowed

Reply:
Value allowed in this case is the value the permission registry expects when setting any ethereum contract call and so even if we are setting permissions for an ERC20 token it still takes this value to mean native token (ethereum) value allowed. As in the value we set in `value` along with `to` and `data`.
This means we essentially set unlimited value transfers on the ERC20 token since the default has no limit. This is what the other `addERC20Limit` function we will add later is for.

Reply:
Thanks for the context. When I read "we should make the default be any amount" on the issue description I thought it was the max_amount. Fixed!